Updates for building with NCrystal support (and fix CI)

tkittel opened the pull request:
# Description

This PR has two NCrystal related fixes, which ensures:

1. The code will keep building once NCrystal 4.0.0 is out in a few weeks (it will remain compatible with older NCrystal releases, the fix is simply to remove a spurious include statement which was never needed).
2. That the CMake layer can find NCrystal, even when NCrystal was installed via a Python tool like pip.

I also made a minimal related documentation change, since NCrystal should in general be installed from conda or pip these days, so there is no need for any post-install setup hassle.

This issue does not fix any open issues, since NCrystal release 4.0.0 is not out yet :-)

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have run [clang-format](https://docs.openmc.org/en/latest/devguide/styleguide.html#automatic-formatting) (version 15) on any C++ source files (if applicable)
- [x] I have followed the [style guidelines](https://docs.openmc.org/en/latest/devguide/styleguide.html#python) for Python source files (if applicable)
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
<!--
While tests will automatically be checked by CI, it is good practice to
ensure that they pass locally first. See instructions here:
https://docs.openmc.org/en/latest/devguide/tests.html
-->

(I checked everything, but of course a few were not applicable).

tkittel commented:
The cpp-linter failure must surely be unrelated to this PR, since all I did was to remove a single include statement!

ebknudsen commented:
@tkittel - I took a look at the raw logs - it appears that the problem is in ll:60
<code>2025-01-23T09:02:09.7763192Z ##[notice]File include/openmc/ncrystal_interface.h does not conform to Custom style guidelines. (lines 60, 61, 62, 64, 65, 66, 69, 70, 71)</code>
You didn't change those lines so the linter directives have probably changed. Don't know _what_ the problem is though.

tkittel commented:
@ebknudsen yeah thanks, that's how far I got as well :-)

If there was some sort of hint of what needs to be changed, I would not mind to go and do a quick fix. However, it really *is* unrelated to this PR, and I would prefer to handle that in a separate PR.

tkittel commented:
Well, I took a look at the workflow and figured out that one can try to run `clang-format --style=file <somefile.h>` to see what it has to be like. So I fixed the linter, and also rebased for good measure.

tkittel commented:
@pshriwise so I added a fix now for the CI like mentioned on #3269 .

I am in principle on vacation now, but if the CI all checks out in this PR, I recommend merging it to fix your CI :-)

(if it does NOT all check out, I will try to fix it)

tkittel commented:
Looks like a few more issues to resolve, trying to have a look.

tkittel commented:
@pshriwise assuming the last non-NCrystal CI jobs also finish fine, I believe this PR is now ready. In summary:

- It stops including a header file which was marked obsolete in NCrystal 4.0.0 (released yesterday).
- It allows OpenMC CMake code to find NCrystal, even when NCrystal was installed via `pip install ncrystal`.
- It fixes the current CI failures, by removing usage of features like `ncrystal-config --setup` which were obsoleted in the NCrystal 4.0.0. Instead it simply uses a pip installed ncrystal, where no special post-installation steps are needed.
- It also updates a CI reference output file to be correct for NCrystal 4 (there were very minor wiggles in some output - most likely due to some tiny bug fixes in the new release (altering inelastic cross sections by 1e-4).
